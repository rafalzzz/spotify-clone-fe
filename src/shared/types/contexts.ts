import { NotificationInstance } from 'antd/es/notification/interface';

export type TNotificationContext = {
  api: NotificationInstance;
};
