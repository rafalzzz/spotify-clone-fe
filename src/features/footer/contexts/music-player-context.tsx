'use client';
import React, {
  createContext,
  useState,
  useMemo,
  useContext,
  useRef,
  PropsWithChildren,
  FC,
} from 'react';

import { TMusicPlayerContext } from '../types';

const MusicPlayerContext = createContext<TMusicPlayerContext | undefined>(undefined);

export const MusicPlayerContextProvider: FC<PropsWithChildren> = ({ children }) => {
  const [currentTime, setCurrentTime] = useState(0);
  const [isShuffle, setIsShuffle] = useState(false);
  const [isLoop, setIsLoop] = useState(false);

  const ref = useRef<HTMLAudioElement>(null);

  const memoizedValue = useMemo(
    () => ({
      ref,
      currentTime,
      isShuffle,
      isLoop,
      setIsShuffle,
      setIsLoop,
      setCurrentTime,
    }),
    [currentTime, isLoop, isShuffle],
  );

  return (
    <MusicPlayerContext.Provider value={memoizedValue}>{children}</MusicPlayerContext.Provider>
  );
};

export const useMusicPlayerContext = (): TMusicPlayerContext => {
  const ctx = useContext(MusicPlayerContext);
  if (ctx === undefined) {
    throw new Error('useMusicPlayerContext must be used within a MusicPlayerContextProvider');
  }

  return ctx;
};
