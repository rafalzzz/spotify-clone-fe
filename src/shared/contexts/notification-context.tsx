'use client';
import React, { createContext, useMemo, useContext, FC, PropsWithChildren } from 'react';

import { TNotificationContext } from '@/types/contexts';

const NofiticationContext = createContext<TNotificationContext | undefined>(undefined);

export const NotificationContextProvider: FC<PropsWithChildren<TNotificationContext>> = ({
  children,
  api,
}) => {
  const memoizedValue = useMemo(
    () => ({
      api,
    }),
    [api],
  );

  return (
    <NofiticationContext.Provider value={memoizedValue}>{children}</NofiticationContext.Provider>
  );
};

export const useNotificationContext = (): TNotificationContext => {
  const ctx = useContext(NofiticationContext);
  if (ctx === undefined) {
    throw new Error('useNotificationContext must be used within a NotificationContextProvider');
  }

  return ctx;
};
