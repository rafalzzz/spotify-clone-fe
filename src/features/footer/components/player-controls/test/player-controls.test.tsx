import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { MusicPlayerContextProvider } from '@/footer/contexts/music-player-context';

import { PlayerControls } from '../';

describe('PlayerControls', () => {
  it('render component without error', () => {
    const screen = render(
      <MusicPlayerContextProvider>
        <PlayerControls />
      </MusicPlayerContextProvider>,
    );
    expect(screen).toMatchSnapshot();
  });
});
