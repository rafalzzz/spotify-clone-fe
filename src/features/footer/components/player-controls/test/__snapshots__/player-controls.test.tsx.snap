// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerControls render component without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="player-controls"
      >
        <div
          class="player-buttons"
        >
          <button
            class="custom-icon-button"
            role="button"
          >
            <svg
              class="svg-icon"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="svg-icon-path"
                d="m4.5 6.8.7-.8C4.1 4.7 2.5 4 .9 4v1c1.3 0 2.6.6 3.5 1.6l.1.2zm7.5 4.7c-1.2 0-2.3-.5-3.2-1.3l-.6.8c1 1 2.4 1.5 3.8 1.5V14l3.5-2-3.5-2v1.5zm0-6V7l3.5-2L12 3v1.5c-1.6 0-3.2.7-4.2 2l-3.4 3.9c-.9 1-2.2 1.6-3.5 1.6v1c1.6 0 3.2-.7 4.2-2l3.4-3.9c.9-1 2.2-1.6 3.5-1.6z"
              />
            </svg>
          </button>
          <button
            class="custom-icon-button"
            role="button"
          >
            <svg
              class="svg-icon"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="svg-icon-path"
                d="M13 2.5L5 7.119V3H3v10h2V8.881l8 4.619z"
              />
            </svg>
          </button>
          <button
            class="custom-icon-button"
            role="button"
          >
            <span
              aria-label="play-circle"
              class="anticon anticon-play-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="play-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
                />
              </svg>
            </span>
          </button>
          <button
            class="custom-icon-button"
            role="button"
          >
            <svg
              class="svg-icon"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="svg-icon-path"
                d="M11 3v4.119L3 2.5v11l8-4.619V13h2V3z"
              />
            </svg>
          </button>
          <button
            class="custom-icon-button"
            role="button"
          >
            <svg
              class="svg-icon"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="svg-icon-path"
                d="M5.5 5H10v1.5l3.5-2-3.5-2V4H5.5C3 4 1 6 1 8.5c0 .6.1 1.2.4 1.8l.9-.5C2.1 9.4 2 9 2 8.5 2 6.6 3.6 5 5.5 5zm9.1 1.7-.9.5c.2.4.3.8.3 1.3 0 1.9-1.6 3.5-3.5 3.5H6v-1.5l-3.5 2 3.5 2V13h4.5C13 13 15 11 15 8.5c0-.6-.1-1.2-.4-1.8z"
              />
            </svg>
          </button>
        </div>
        <div
          class="music-progress-bar"
        >
          <audio />
          <button
            class="duration duration--text-align-end"
            data-testid="duration-button"
            disabled=""
          >
            0:00
          </button>
          <div
            class="progress-bar"
          >
            <input
              class="progress-bar__input"
              max="0"
              min="0"
              step="0.01"
              type="range"
              value="0"
            />
            <span
              class="progress-bar__span-thumb"
              style="left: -3px;"
            />
          </div>
          <button
            class="duration duration--text-align-start"
            data-testid="duration-button"
          >
            0:00
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="player-controls"
    >
      <div
        class="player-buttons"
      >
        <button
          class="custom-icon-button"
          role="button"
        >
          <svg
            class="svg-icon"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="svg-icon-path"
              d="m4.5 6.8.7-.8C4.1 4.7 2.5 4 .9 4v1c1.3 0 2.6.6 3.5 1.6l.1.2zm7.5 4.7c-1.2 0-2.3-.5-3.2-1.3l-.6.8c1 1 2.4 1.5 3.8 1.5V14l3.5-2-3.5-2v1.5zm0-6V7l3.5-2L12 3v1.5c-1.6 0-3.2.7-4.2 2l-3.4 3.9c-.9 1-2.2 1.6-3.5 1.6v1c1.6 0 3.2-.7 4.2-2l3.4-3.9c.9-1 2.2-1.6 3.5-1.6z"
            />
          </svg>
        </button>
        <button
          class="custom-icon-button"
          role="button"
        >
          <svg
            class="svg-icon"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="svg-icon-path"
              d="M13 2.5L5 7.119V3H3v10h2V8.881l8 4.619z"
            />
          </svg>
        </button>
        <button
          class="custom-icon-button"
          role="button"
        >
          <span
            aria-label="play-circle"
            class="anticon anticon-play-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="play-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
              />
            </svg>
          </span>
        </button>
        <button
          class="custom-icon-button"
          role="button"
        >
          <svg
            class="svg-icon"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="svg-icon-path"
              d="M11 3v4.119L3 2.5v11l8-4.619V13h2V3z"
            />
          </svg>
        </button>
        <button
          class="custom-icon-button"
          role="button"
        >
          <svg
            class="svg-icon"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="svg-icon-path"
              d="M5.5 5H10v1.5l3.5-2-3.5-2V4H5.5C3 4 1 6 1 8.5c0 .6.1 1.2.4 1.8l.9-.5C2.1 9.4 2 9 2 8.5 2 6.6 3.6 5 5.5 5zm9.1 1.7-.9.5c.2.4.3.8.3 1.3 0 1.9-1.6 3.5-3.5 3.5H6v-1.5l-3.5 2 3.5 2V13h4.5C13 13 15 11 15 8.5c0-.6-.1-1.2-.4-1.8z"
            />
          </svg>
        </button>
      </div>
      <div
        class="music-progress-bar"
      >
        <audio />
        <button
          class="duration duration--text-align-end"
          data-testid="duration-button"
          disabled=""
        >
          0:00
        </button>
        <div
          class="progress-bar"
        >
          <input
            class="progress-bar__input"
            max="0"
            min="0"
            step="0.01"
            type="range"
            value="0"
          />
          <span
            class="progress-bar__span-thumb"
            style="left: -3px;"
          />
        </div>
        <button
          class="duration duration--text-align-start"
          data-testid="duration-button"
        >
          0:00
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
